File: AkkaSTMExample/src/main/scala/org/akka/essentials/scala/stm/stockticker/example/StockUpdater.scala
package org.akka.essentials.scala.stm.stockticker.example


class StockUpdater(stock: Stock) extends Runnable {

  var countDown = 5

  override def run(): Unit = {
    while (countDown > 0) {
      Thread.sleep(55)
      val x: String = Thread.currentThread().getName
      stock.price.send(_ + 10)
      println("Quote update by thread (" + x + "), current price " + stock.price.get)
      countDown = countDown - 1
    }
  }
}

File: AkkaSTMExample/src/main/java/org/akka/essentials/stm/stockticker/example/StockUpdater.java
package org.akka.essentials.stm.stockticker.example;

import akka.japi.Function;

public class StockUpdater implements Runnable {

	private int countDown = 5;
	private Stock stock;

	public StockUpdater(Stock inStock) {
		stock = inStock;
	}

	public void run() {
		while (countDown > 0) {
			try {
				Thread.sleep(75);
			} catch (InterruptedException e) {
			}
			String x = Thread.currentThread().getName();
			stock.getPrice().send(new Function<Float, Float>() {
				public Float apply(Float i) {
					return i + 10;
				}
			});
			System.out.println("Quote update by thread (" + x
					+ "), current price " + stock.getPrice().get());
			countDown = countDown - 1;
		}
	}
}
